sdx-submitter.go

```diff
@@ -0,0 +1,168 @@
+package main
+import(
+		"fmt"
+		"flag"
+		"github.com/streadway/amqp"
+		"os"
+		"io/ioutil"
+)
+
+func main() {
+
+	// Read a source file and place it on a rabbit topic exchange
+	var url, exchange, routing_key,    encryption_key,    signing_key, name, password string
+	var port, host , queue, vhost, message_path_and_file string
+	var msg_body []byte
+	var err error
+
+	getArguments(&name, &password, &port, &host, &url, &queue, &exchange, &routing_key, &encryption_key, &signing_key, &vhost, &message_path_and_file)
+
+	msg_body = getBody(message_path_and_file)
+
+	// If encyrpt specified then encrypt
+	// if sign specified then sign
+
+	sendToRabbit(url, exchange, queue, routing_key, err, msg_body)
+	
+}
+
+func getArguments(name *string,
+	password *string,
+	port *string,
+	host *string,
+	url *string,
+	queue *string,
+	exchange *string,
+	routing_key *string,
+	encryption_key *string,
+	signing_key *string,
+	vhost *string,
+	message_path_and_file *string)  {
+	// define the command line parameters
+	flag.StringVar(name, "n", "", "name of the rabbit user")
+	flag.StringVar(password, "p", "", "password of the rabbit user")
+	flag.StringVar(port, "rport", "", "port used to connect to rabbit")
+	flag.StringVar(host, "rhost", "localhost", "hostname used to connect to rabbit")
+	flag.StringVar(vhost, "v", "", "vhostname used to connect to rabbit")
+	flag.StringVar(url, "u", "", "url connection string ")
+	flag.StringVar(queue, "q", "", "name of the rabbit queue")
```

Comment:
You can hardcode the queue name, exchange and routing key as they should be standard across envs for the eq queue (if they're not then that's something separate we need to look at outside of this! :-) )

Reply:
A lot of the other "standard" type settings could be assumed to be coming from the environment (possibly sourced through a config file) or from a file directly (like <name-of-binary>.cfg) or something - reduces the footprint of the command line for the user and isn't likely to alter between multiple invocations in a shell.

Reply:
current implementation takes values from command line , if none present it uses environment variables as already defined for sdc common, then if not present then it uses a hardcoded default